maven-core/src/main/java/org/apache/maven/session/RepositorySystemSessionExtender.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.session;
+
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.eclipse.aether.RepositorySystemSession.SessionBuilder;
+import org.eclipse.aether.repository.AuthenticationSelector;
+import org.eclipse.aether.repository.MirrorSelector;
+import org.eclipse.aether.repository.ProxySelector;
+
+/**
+ * Component able to "extend" session in some way.
+ *
+ * @since 4.0.0
+ */
+public interface RepositorySystemSessionExtender {
+    void extend(
+            MavenExecutionRequest mavenExecutionRequest,
+            SessionBuilder sessionBuilder,
+            MirrorSelector mirrorSelector,
```

Comment:
Those 3 last parameters looks really coming out of nowhere.
In particular, I think they can be accessed from the `sessionBuilder` as it has getters for those and they are set earlier, so I'd remove those 3 params.

Reply:
sessionBuilder does not have these, you told to remove them :smile:  so we are going back and forth... anyway, this is strictly "internal" component (see package), and let me push more.